opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/MonitoringLocationRestService.java

```diff
@@ -88,10 +88,10 @@ protected Class<OnmsMonitoringLocation> getQueryBeanClass() {
 
     @Override
     protected CriteriaBuilder getCriteriaBuilder(UriInfo uriInfo) {
-        final CriteriaBuilder builder = new CriteriaBuilder(OnmsMonitoringLocation.class);
+        CriteriaBuilder builder = new CriteriaBuilder(OnmsMonitoringLocation.class);
 
-        // Order by location name by default
-        builder.orderBy("locationName").asc();
+        // Order by location name by default and remove limit on list
+        builder = builder.orderBy("locationName").asc().limit(0);
```

Comment:
Unrelated changes, should be removed from this PR.

Reply:
I have rolled theses changes back, my apologies